// Copyright 2018 Intel Corporation
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
// -----------------------------------------------------------------------------

syntax = "proto3";


message User {
    // User's unique public key
    string public_key = 1;

    // A string with the User was registered, as a Unix UTC timestamp
    string name = 2;

    // Approximately when the user was registered, as a Unix UTC timestamp
    uint64 timestamp = 3;

    // Water quota granted to the user
    double quota = 4;
}


message UserContainer {
    repeated User entries = 1;
}
